ledger/participant-integration-api/src/main/scala/platform/index/ReadOnlySqlLedger.scala

```diff
@@ -65,9 +67,25 @@ private[platform] object ReadOnlySqlLedger {
 
     override def acquire()(implicit context: ResourceContext): Resource[ReadOnlySqlLedger] = {
       val ledgerEndCache = MutableLedgerEndCache()
+      val storageBackendFactory = StorageBackendFactory.of(DbType.jdbcType(jdbcUrl))
```

Comment:
`DbType.jdbcType()` can throw an exception which currently will not be enclosed in the returned `Resource` because it's outside of the `for` comprehesion below.
I would consider moving it to the `for` below to allow recovering the resource.

Reply:
I don't think there is any benefit to move that inside: no recovery possible if that function throws an exception.